src/pages/Proposal/VoteModal/index.tsx

```diff
@@ -49,22 +48,17 @@ const VoteModal: React.FC<IVoteModal> = ({
   }
 
   const handleOnSubmit = async ({ reason }: { reason: string }) => {
-    try {
-      const transactionReceipt = await vote(reason);
-      handleClose();
-      openSuccessfulTransactionModal({
+    await handleTransactionMutation({
+      mutate: () => vote(reason),
+      successTransactionModalProps: transactionReceipt => ({
         title: successModalTitle,
         content: t('vote.pleaseAllowTimeForConfirmation'),
         transactionHash: transactionReceipt.transactionHash,
-      });
-    } catch (error) {
-      let { message } = error as Error;
-      if (error instanceof VError) {
-        message = formatVErrorToReadableString(error);
-      }
-      toast.error({ message });
-    }
+      }),
+    });
+    handleClose();
```

Comment:
This should be part of the `mutate` function, so that's it's called on success only (otherwise the modal will also close when an error occurred)